packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -217,7 +350,60 @@ export const prepareStakingAccountKit = (baggage, makeRecorderKit, zcf) => {
           if (!result) throw Fail`Failed to delegate.`;
           return tryDecodeResponse(result, MsgDelegateResponse.fromProtoMsg);
         },
+        /**
+         * Only `bondDenom` deposits accepted until #9211, #9063
+         * @param {Payment} payment
+         * @param {{ timeoutTimestamp: bigint }} [opts]
+         * @returns {Promise<{ sequence: number }>}
+         */
+        async deposit(payment, opts) {
+          const brand = await E(payment).getAllegedBrand();
+          const issuer = this.state.brandToIssuer.get(brand);
+          issuer || Fail`Unknown Issuer for Brand ${brand}.`;
+
+          const amount = await E(issuer).getAmountOf(payment);
+          !AmountMath.isEmpty(amount) ||
+            Fail`Payment amount must be greater than 0.`;
 
+          const { localAccount } = this.state;
+          trace('Depositing funds to LCA');
+          // XXX consider adding exposing an interface to withdraw / send
+          // messages from the LCA (e.g., withdraw funds)
+          await E(localAccount).deposit(payment);
+
+          const timeoutTimestamp =
+            opts?.timeoutTimestamp ??
+            (await this.facets.helper.getTimeoutTimestamp());
+
+          trace('Transferring funds to ICA');
+          /**
+           * // TODO can we infer `/ibc.applications.transfer.v1.MsgTransferResponse`?
+           * @type {unknown[]}
```

Comment:
My current (unsuccessful) attempt at this:

<details>
<summary>
packages/cosmic-proto/src/helpers.ts
</summary>

```diff
git diff packages/cosmic-proto/src/helpers.ts
diff --git a/packages/cosmic-proto/src/helpers.ts b/packages/cosmic-proto/src/helpers.ts
index ca192be60..00ebfea30 100644
--- a/packages/cosmic-proto/src/helpers.ts
+++ b/packages/cosmic-proto/src/helpers.ts
@@ -1,6 +1,15 @@
-import type { QueryAllBalancesRequest } from './codegen/cosmos/bank/v1beta1/query.js';
-import type { MsgSend } from './codegen/cosmos/bank/v1beta1/tx.js';
-import type { MsgDelegate } from './codegen/cosmos/staking/v1beta1/tx.js';
+import type {
+  QueryAllBalancesRequest,
+  QueryAllBalancesResponse,
+} from './codegen/cosmos/bank/v1beta1/query.js';
+import type {
+  MsgSend,
+  MsgSendResponse,
+} from './codegen/cosmos/bank/v1beta1/tx.js';
+import type {
+  MsgDelegate,
+  MsgDelegateResponse,
+} from './codegen/cosmos/staking/v1beta1/tx.js';
 import { RequestQuery } from './codegen/tendermint/abci/types.js';
 import type { Any } from './codegen/google/protobuf/any.js';
 import {
@@ -13,17 +22,32 @@ import {
  * `unknown` but it's actually this. The `value` string is a base64 encoding of
  * the bytes array.
  */
-export type AnyJson = { typeUrl: string; value: string };
-
-// TODO codegen this by modifying Telescope
+export type AnyJson = {
+  typeUrl: string;
+  value: string;
+};
 export type Proto3Shape = {
   '/cosmos.bank.v1beta1.MsgSend': MsgSend;
+  '/cosmos.bank.v1beta1.MsgSendResponse': MsgSendResponse;
   '/cosmos.bank.v1beta1.QueryAllBalancesRequest': QueryAllBalancesRequest;
+  '/cosmos.bank.v1beta1.QueryAllBalancesResponse': QueryAllBalancesResponse;
   '/cosmos.staking.v1beta1.MsgDelegate': MsgDelegate;
+  '/cosmos.staking.v1beta1.MsgDelegateResponse': MsgDelegateResponse;
   '/ibc.applications.transfer.v1.MsgTransfer': MsgTransfer;
   '/ibc.applications.transfer.v1.MsgTransferResponse': MsgTransferResponse;
 };

+export type ResponseType<T> =
+  T extends '/ibc.applications.transfer.v1.MsgTransfer'
+    ? '/ibc.applications.transfer.v1.MsgTransferResponse'
+    : T extends '/cosmos.bank.v1beta1.MsgSend'
+      ? '/cosmos.bank.v1beta1.MsgSendResponse'
+      : T extends '/cosmos.staking.v1beta1.MsgDelegate'
+        ? '/cosmos.staking.v1beta1.MsgDelegateResponse'
+        : T extends '/cosmos.bank.v1beta1.QueryAllBalancesRequest'
+          ? '/cosmos.bank.v1beta1.QueryAllBalancesResponse'
+          : never;
+
 /**
  * The encoding introduced in Protobuf 3 for Any that can be serialized to JSON.
  *

```
</details>

<details>
<summary>
packages/vats/src/localchain.js
</summary>

```diff
git diff packages/vats/src/localchain.js
diff --git a/packages/vats/src/localchain.js b/packages/vats/src/localchain.js
index 5a07e8ad3..6ab60c379 100644
--- a/packages/vats/src/localchain.js
+++ b/packages/vats/src/localchain.js
@@ -5,6 +5,8 @@ import { AmountShape } from '@agoric/ertp';

 const { Fail, bare } = assert;

+/** @import {TypedJson, ResponseType, Proto3Shape} from '@agoric/cosmic-proto'; */
+
 /**
  * @typedef {{
  *   system: import('./types.js').ScopedBridgeManager;
@@ -82,8 +84,9 @@ const prepareLocalChainAccount = zone =>
         return E(purse).withdraw(amount);
       },
       /**
-       * @param {import('@agoric/cosmic-proto').TypedJson<unknown>[]} messages
-       * @returns {Promise<import('@agoric/cosmic-proto').TypedJson[]>}
+       * @template {keyof Proto3Shape} T
+       * @param {TypedJson<T>[]} messages
+       * @returns {Promise<TypedJson<ResponseType<T>>[]>}
        */
       async executeTx(messages) {
         const { address, powers } = this.state;
@@ -178,8 +181,8 @@ const prepareLocalChain = (zone, makeAccount) =>
          * the query fails. Otherwise, return the response as a JSON-compatible
          * object.
          *
-         * @param {import('@agoric/cosmic-proto').TypedJson} request
-         * @returns {Promise<import('@agoric/cosmic-proto').TypedJson>}
+         * @param {TypedJson} request
+         * @returns {Promise<TypedJson>}
          */
         async query(request) {
           const requests = harden([request]);
@@ -197,11 +200,11 @@ const prepareLocalChain = (zone, makeAccount) =>
          * system error, will return all results to indicate their success or
          * failure.
          *
-         * @param {import('@agoric/cosmic-proto').TypedJson[]} requests
+         * @param {TypedJson[]} requests
          * @returns {Promise<
          *   {
          *     error?: string;
-         *     reply: import('@agoric/cosmic-proto').TypedJson;
+         *     reply: TypedJson;
          *   }[]
          * >}
          */
```
</details>

Reply:
https://github.com/Agoric/agoric-sdk/pull/9351